internal/execution/request.go

```diff
@@ -0,0 +1,74 @@
+package execution
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/zitadel/logging"
+
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/query"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+type ContextInfoRequest struct {
+	FullMethod string      `json:"fullMethod,omitempty"`
+	InstanceID string      `json:"instanceID,omitempty"`
+	OrgID      string      `json:"orgID,omitempty"`
+	ProjectID  string      `json:"projectID,omitempty"`
+	UserID     string      `json:"userID,omitempty"`
+	Request    interface{} `json:"request,omitempty"`
+}
+
+func CallTargetsRequest(ctx context.Context,
```

Comment:
The logic feels very complex, I needed to read it multiple times and am not really sure if my mental model is correct.

I would try to create a struct like follows and implement the logic in a single place instead of multiple functions.

```
type Targets []*Target

type Target struct{
    *query.Target
}

func (t *Target) Call() ...
```

Reply:
Remodelled the logic